### PR TITLE
Allow use of default MPI Communicator with Fortran API

### PR DESCRIPTION
**Pull request by andrewcoughtrie**
Although Vernier will use a default MPI communicator when called from the C++ API this has not been passed through to the Fortran API. Have added the capability here. 

I haven't been able to get the testing working using pFUnit, I'm not sure why, for some reason it hangs. I do know that pFUnit controls the MPI communicator and duplicates it, it does state in the source not to use MPI_COMM_WORLD outside the main pFUnit program [MpiContext.F90#L65](https://github.com/Goddard-Fortran-Ecosystem/pFUnit/blob/main/src/pfunit/core/MpiContext.F90#L65) so perhaps they are doing some thing that doesn't play well if we try to use MPI_COMM_WORLD for Vernier as well.

As it is I've created a simple system test to check the call without an argument works, we perhaps need a better mechanism for testing this at some point.

I've also done a bit of tidying up of all the test names.

### CODE REVIEW

File: tests/system_tests/f/test_initialisation.F90 [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  ! Initialise Vernier with the default communicator (i.e. pass no communicator)
+  call vernier_init()
+
+  call vernier_start(hash_out, "test_initialisation")
+  ! Do some work here
+  call sleep(1)
+  call vernier_stop(hash_out)
+
+  Call vernier_finalize()
+
```
Comment (mo-mglover) [inline tests/system_tests/f/test_initialisation.F90:30]:
```suggestion
call mpi_finalize(ierr)
```

File: tests/system_tests/f/test_initialisation.F90 [hunk 2]
```diff
@@ -0,0 +1,32 @@
+!-------------------------------------------------------------------------------
+! (c) Crown copyright 2025 Met Office. All rights reserved.
+! The file LICENCE, distributed with this code, contains details of the terms
+! under which the code may be used.
+!-------------------------------------------------------------------------------
+
```
Comment (mo-mglover) [inline tests/system_tests/f/test_initialisation.F90:6]: Might be worth a comment block to say why this is a system test rather than a unit test.
(pFUnit wanting to dictate the communicator handle.)

**Review by mo-mglover — CHANGES_REQUESTED** [review]
Looks good to me.  I agree that the `optional` argument approach is the right way to do this.

I've made a couple of inline comments.
[Noticed a few capitalisation issues, too, but I haven't nit-picked about those!]

Also noted, just for the papertrail, that there's a number of other bits of tidying up mixed in, such as a `cmake` function to facilitate adding new Fortran pfunit tests and removal of trailing whitespace.

File: tests/system_tests/f/test_initialisation.F90 (hunk 2 above)
Comment (andrewcoughtrie) [reply tests/system_tests/f/test_initialisation.F90]: Added a comment to describe the reasoning for the test.

**Review by mo-mglover — APPROVED** [review]
